### PR TITLE
Takeaway challenge

### PR DESCRIPTION
**Pull request by NadiaAiraf**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,66 +14,59 @@ Takeaway Challenge
 
  ```
 
-Instructions
--------
+This repository contains the week 2 Makers Academy homework challenge. The program is run on the command line and allows you to see everything on a takeaway menu, place orders from that menu and complete the order and have a message confirming your order and a delivery time sent to your phone using Twilio.
```
Comment (soph-g) [inline README.md]: Good detail in the README.
Could have added a diagram of your application design to provide even more information.
RE commits - I would encourage you to make your commits smaller - make a commit for every change. A change is: writing a test, using TDD to write code until the test passes, refactor if appropriate.

File: lib/message.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require 'twilio-ruby'
+
+class Message
+  attr_reader :account_sid, :auth_token,
+              :client, :from, :to
+
+  def initialize(client = Twilio::REST::Client.new('ACbad1cc1fe45037b01a7fd207fb7f270c', '2223f6422dca48314885d82d94b3d005'))
```
Comment (soph-g) [inline lib/message.rb]: Data like this should be stored in environment variables or a config file that is not committed - to protect the data. 
Please look at removing this data from GitHub and using environment variables to store and access it locally.

File: lib/takeaway.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+require_relative 'message'
+
+class TakeAway
+  attr_reader :menu, :basket, :order_total, :message
```
Comment (soph-g) [inline lib/takeaway.rb]: Does message message need to be publicly available?

File: lib/takeaway.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            "Crispy Shredded Beef" => 5.90,
+            "Egg Fried Rice" => 3.00,
+            "Fried Duck with Black Bean Sauce" => 6.40
+            }
+    @basket = {}
+    @order_total = 0
+    @message = message
+  end
+
+  def read_menu
```
Comment (soph-g) [inline lib/takeaway.rb]: This class has a few responsibilities. It
- Holds the menu information
- Manages the order

File: lib/takeaway.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    menu.each { |k, v| puts "#{k}: £#{'%.2f' % v}" }
+  end
+
+  def add_dish(dish, number = 1)
+    # binding.pry
```
Comment (soph-g) [inline lib/takeaway.rb]: Remove commented out code.

File: send_sms.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
+# Download the twilio-ruby library from twilio.com/docs/libraries/ruby
```
Comment (soph-g) [inline send_sms.rb]: Remove commented out code.

File: send_sms.rb [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+#     from: from,
+#     to: to,
+#     body: "you've placed a takeaway order, it will arrive in 50mins"
+#     )
+#   end
+# end
+
+require 'twilio-ruby'
+
+account_sid = 'ACbad1cc1fe45037b01a7fd207fb7f270c'
```
Comment (soph-g) [inline send_sms.rb]: Use environment variables to store this data locally. Please remove it from GitHub (account SID, auth token and phone numbers) and use environment variables instead.

File: send_sms.rb [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+from = '+447480485499' # Your Twilio number
+to = '+447828166080' # Your mobile phone number
+
+client.messages.create(
+from: from,
+to: to,
+body: "Hey friend!"
+)
+
+# menu class:
```
Comment (soph-g) [inline send_sms.rb]: Remove commented out code.

File: spec/message_spec.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+require 'message'
+
+describe Message do
+  let(:mockMessages) { double :messages, create: nil }
+  let(:mockTwilio) { double :twilio, messages: mockMessages }
+  subject { described_class.new(mockTwilio) }
```
Comment (soph-g) [inline spec/message_spec.rb]: Good set up of doubles.

File: spec/takeaway_spec.rb [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    it 'returns a display of everything on the menu (currently hardcoded)' do
+      expect(subject.read_menu).to eq({ "Sweet & Sour Chicken Balls" => 3.99,
+                                        "Chicken Chowmein" => 4.50,
+                                        "Crispy Shredded Beef" => 5.90,
+                                        "Egg Fried Rice" => 3.00,
+                                        "Fried Duck with Black Bean Sauce" => 6.40 })
+    end
+  end
+
+  describe '#select_dish' do
```
Comment (soph-g) [inline spec/takeaway_spec.rb]: Good grouping of tests by method

File: spec/takeaway_spec.rb [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    it 'adds a dish to my basket when I select it' do
+      subject.add_dish('Egg Fried Rice')
+      expect(subject.basket).to eq({ "Egg Fried Rice" => 1 })
+    end
+
+    it 'adds the same dish multiple times to my basket' do
+      subject.add_dish('Egg Fried Rice', 4)
```
Comment (soph-g) [inline spec/takeaway_spec.rb]: Can I update the quantity of an item? e.g. add 2 egg fried rice, then add another to order 3?

Comment (NadiaAiraf) [reply spec/takeaway_spec.rb]: Yes :) that was done in the next test on from this

**Pull request closed without merging**